cmd/snap/cmd_debug_state.go

```diff
@@ -227,6 +234,201 @@ func (c *cmdDebugState) showIsSeeded(st *state.State) error {
 	return nil
 }
 
+// connState is the connection state stored by InterfaceManager
+type connState struct {
+	Auto             bool                   `json:"auto,omitempty" yaml:"auto"`
+	ByGadget         bool                   `json:"by-gadget,omitempty" yaml:"by-gadget"`
+	Interface        string                 `json:"interface,omitempty" yaml:"interface"`
+	Undesired        bool                   `json:"undesired,omitempty" yaml:"undesired"`
+	StaticPlugAttrs  map[string]interface{} `json:"plug-static,omitempty" yaml:"plug-static,omitempty"`
+	DynamicPlugAttrs map[string]interface{} `json:"plug-dynamic,omitempty" yaml:"plug-dynamic,omitempty"`
+	StaticSlotAttrs  map[string]interface{} `json:"slot-static,omitempty" yaml:"slot-static,omitempty"`
+	DynamicSlotAttrs map[string]interface{} `json:"slot-dynamic,omitempty" yaml:"slot-dynamic,omitempty"`
+	// TODO: hotplug
+}
+
+type connectionInfo struct {
+	PlugSnap string
+	PlugName string
+	SlotSnap string
+	SlotName string
+
+	connState
+}
+
+type byPlug []*connectionInfo
+
+func (c byPlug) Len() int      { return len(c) }
+func (c byPlug) Swap(i, j int) { c[i], c[j] = c[j], c[i] }
+func (c byPlug) Less(i, j int) bool {
+	a := c[i]
+	b := c[j]
+
+	if a.PlugSnap < b.PlugSnap {
+		return true
+	}
+
+	if a.PlugSnap == b.PlugSnap {
+		if a.PlugName < b.SlotName {
+			return true
+		}
+	}
+	return false
+}
+
+func parsePlugOrSlot(plugOrSlot string) (snap, name string, err error) {
```

Comment:
Can you reuse anything from https://github.com/snapcore/snapd/blob/master/cmd/snap/interfaces_common.go so that there's one less copy of this code?

Reply:
Done.